aggregator/src/aggregator.rs

```diff
@@ -1397,14 +1404,16 @@ impl VdafOps {
         C: Clock,
         Q: UploadableQueryType,
     {
-        // Shorthand function for generating an Error::ReportRejected with proper parameters.
+        // Shorthand function for generating an Error::ReportRejected with proper parameters and
+        // recording it in the report_writer.
         let reject_report = |reason| {
-            Arc::new(Error::ReportRejected(
-                *task.id(),
-                *report.metadata().id(),
-                *report.metadata().time(),
-                reason,
-            ))
+            let report_id = *report.metadata().id();
+            let report_time = *report.metadata().time();
+            async move {
+                let rejection = ReportRejection::new(*task.id(), report_id, report_time, reason);
+                report_writer.write_report(Err(rejection)).await?;
```

Comment:
I think a couple of times now we've floated the idea of successful uploads returning OK immediately after they're enqueued. One example: https://isrg.slack.com/archives/C0167LT4C73/p1701364685606769?thread_ts=1701364378.124579&cid=C0167LT4C73.

Should we go ahead and take this change? That'll solve your comment as well, since we won't ever be waiting on batches flushing to the DB anyway.

Reply:
https://github.com/divviup/janus/pull/2508/commits/9a12677657c2a8112fe796c58afa64f63eb03b46 introduces `write_rejection()` which doesn't wait for the tx result.

Notice that the result_tx is now tx/rx'ing a `(ReportResult<C>, Option<ResultSender>)`, i.e. the result sender is optional and in `write_rejection()` we send `None`. I did this because otherwise we'd be creating a fake `Sender` which is immediately dropped, sending `write_batch()` down an unhappy path.

That commit also has some clumsy `sleep()` in tests to wait for the report writer to work. I've replaced that with use of `Runtime` in https://github.com/divviup/janus/pull/2508/commits/37a35937b3f1de35f3d8bbe9c7dae3793056051a